django/db/models/fields/__init__.py

```diff
@@ -772,6 +769,8 @@ def get_db_prep_lookup(self, lookup_type, value, connection,
             return [self.get_db_prep_value(value, connection=connection,
                                            prepared=prepared)]
         elif lookup_type in ('range', 'in'):
+            if not isinstance(value, (list, tuple)):
```

Comment:
Right, so this method and the `process_value` method of `Lookup` need careful scrutiny, because I'm not certain that I haven't broken anything.

Running a section of the test suite (expressions annotations aggregation aggregation_regress expressions_case lookup queries django.contrib.postgres custom_lookups postgres_tests), there's not a single case that IS a list or tuple. I think I put this check in to cater for existing code out in the wild, but it might be unnecessary.

The `range` and `in` lookup types need to "batch process" the rhs arguments. They weren't previously unwrapped correctly with regard to bilateral transformations. I've changed the range/in types to pull out each component of the rhs, apply bilateral transformations, and then send through the `get_db_prep_lookup`. Previously, the entire list was passed through to get_db_prep_lookup.

I don't think custom iterables are going to be a problem here, since each member will be unwrapped earlier in the `process_value` or `batch_process_rhs` methods.

What I do need to do is write a test for bilateral transformations applied to `in` and `range`.